packages/blade/src/components/Input/DropdownInputTriggers/BaseDropdownInputTrigger.tsx

```diff
@@ -0,0 +1,283 @@
+import React from 'react';
+import isEmpty from 'lodash/isEmpty';
+import { BaseInput } from '../BaseInput';
+import { InputChevronIcon } from './InputChevronIcon';
+import type { BaseDropdownInputTriggerProps } from './types';
+import { useDropdown } from '~components/Dropdown/useDropdown';
+import { isReactNative } from '~utils';
+import { getActionListContainerRole } from '~components/ActionList/getA11yRoles';
+import { MetaConstants } from '~utils/metaAttribute';
+import { getTagsGroup } from '~components/Tag/getTagsGroup';
+import type { BladeElementRef } from '~utils/types';
+
+const useControlledDropdownInput = (
+  props: Pick<
+    BaseDropdownInputTriggerProps,
+    'onChange' | 'onSelectionChange' | 'name' | 'value' | 'defaultValue'
+  >,
+): void => {
+  const isFirstRenderRef = React.useRef(true);
```

Comment:
We could have a common util for this just like we have `useUpdateEffect`, we can have `useFirstRender`?

Reply:
done